src/java.base/share/classes/sun/nio/ch/SocketInputStream.java

```diff
@@ -71,6 +72,17 @@ public int read(byte[] b, int off, int len) throws IOException {
         }
     }
 
+    @Override
+    public int read(byte[] b, int off, int len) throws IOException {
+        if (!SocketReadEvent.enabled()) {
+            return implRead(b, off, len);
+        }
+        long start = SocketReadEvent.timestamp();
+        int n = implRead(b, off, len);
+        SocketReadEvent.offer(start, n, sc.remoteAddress(), timeoutSupplier.getAsInt());
```

Comment:
One of the things left over from the original refactoring was to dig into my the event needs the timeout, it would be nice if that could go away.

For now, I'd prefer if the timeout could be captured once as otherwise there is no guarantee that the event will be reported with the same timeout used for the socket read. In other words, add a timeout parameter to implRead so that the method calls implRead with the timeout, avoids calling timeoutSuppler::getAsInt twice.

Reply:
If remoteAddress is an issue then we can look at that, it may be a candidate to a `@Stable` field. That is probably beyond the scope of this PR of course.